gcd/src/main/java/org/spine3/server/storage/datastore/DatastoreStorageFactory.java

```diff
@@ -165,4 +190,70 @@ protected void setDatastore(DatastoreWrapper datastore) {
         this.datastore = datastore;
     }
 
+    /**
+     * Creates new instance of {@link Builder}.
+     */
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    /**
+     * A builder for the {@code DatastoreStorageFactory}.
+     */
+    public static class Builder {
+
+        private Datastore datastore;
+        private boolean multitenant;
+        private ColumnTypeRegistry<DatastoreColumnType> typeRegistry = DatastoreTypeRegistry.defaultInstance();
+
+        private Builder() {
+            // Avoid direct initialization
+        }
+
+        /**
+         * @param datastore the {@link Datastore} to use for the DB interactions
+         * @return self for method chaining
+         */
+        public Builder setDatastore(Datastore datastore) {
+            this.datastore = datastore;
+            return this;
+        }
+
+        /**
+         * Switches the {@code DatastoreStorageFactory} that is being built to be or not to be multitenant.
+         *
+         * @param multitenant {@code true} if the {@code DatastoreStorageFactory} should be multitenant or not
+         * @return self for method chaining
+         */
+        public Builder setMultitenant(boolean multitenant) {
+            this.multitenant = multitenant;
+            return this;
+        }
+
+        /**
+         * Sets a {@link ColumnTypeRegistry} for handling the Entity Columns.
+         *
+         * <p>Default value is {@link DatastoreTypeRegistry#defaultInstance()}.
+         *
+         * @param typeRegistry the type registry containing all the required
+         * {@linkplain org.spine3.server.entity.storage.ColumnType column types} to handle the existing Entity Columns
+         * @return self for method chaining
+         */
+        public Builder setTypeRegistry(ColumnTypeRegistry<DatastoreColumnType> typeRegistry) {
+            this.typeRegistry = typeRegistry;
+            return this;
+        }
+
+        /**
+         * Creates a new instance of {@code DatastoreStorageFactory} with the passed parameters.
+         *
+         * <p>Precondition of a successful build is that the {@code datastore} field has been set.
+         *
+         * @return new instance of {@code DatastoreStorageFactory}
+         */
+        public DatastoreStorageFactory build() {
+            checkNotNull(datastore);
```

Comment:
How come the default value for the `typeRegistry` is not set in `build` method? I assume you don't test this, as well as setting the `null` values. 

Please have `DatastoreStorageFactoryBuilderShould`.